bigtable/internal/testproxy/proxy_test.go

```diff
@@ -0,0 +1,448 @@
+	ctx := context.Background()
+
+	req := &pb.CreateClientRequest{
+		ClientId:   cid,
+		ProjectId:  "client",
+		DataTarget: bta,
+		InstanceId: "instance",
+	}
+
+	_, err := (*client).CreateClient(ctx, req)
+	if err != nil {
+		t.Fatalf("testproxy test: failed to create client: %v", err)
+	}
+
+	// Test
+	_, err = (*client).RemoveClient(ctx, &pb.RemoveClientRequest{
+		ClientId:  cid,
+		CancelAll: true,
+	})
+
+	if err != nil {
+		t.Errorf("testproxy test: RemoveClient() failed: %v", err)
+	}
+}
+
+func TestReadRow(t *testing.T) {
+	ctx := context.Background()
+	req := &pb.ReadRowRequest{
+		TableName: tbl,
+		ClientId:  tpc,
+		RowKey:    rk,
+	}
+
+	resp, err := (*client).ReadRow(ctx, req)
+	if err != nil {
+		t.Fatalf("testproxy test: ReadRow() failed: %v", err)
+	}
+
+	stat := resp.Status.Code
+	if stat != int32(codes.OK) {
+		t.Errorf("testproxy test: ReadRow() didn't return OK")
+	}
+
+	row := resp.Row
+	if string(row.Key) != "row" {
+		t.Errorf("testproxy test: ReadRow() returned wrong row")
+	}
+}
+
+func TestSampleRowKeys(t *testing.T) {
+	ctx := context.Background()
+	req := &pb.SampleRowKeysRequest{
+		ClientId: tpc,
+		Request: &btpb.SampleRowKeysRequest{
+			TableName: tbl,
+		},
+	}
+
+	resp, err := (*client).SampleRowKeys(ctx, req)
+	if err != nil {
+		t.Fatalf("testproxy test: SampleRowKeys() returned error: %v", err)
+	}
+
+	if resp.Status.Code != int32(codes.OK) {
+		t.Errorf("testproxy test: SampleRowKeys() didn't return OK; got %v", resp.Status.Code)
+	}
+
+	if len(resp.Sample) != 1 {
+		t.Errorf("testproxy test: SampleRowKeys() returned wrong number of results; got: %d", len(resp.Sample))
+	}
+}
+
+func TestReadRows(t *testing.T) {
+	ctx := context.Background()
+	req := &pb.ReadRowsRequest{
+		ClientId: tpc,
+		Request: &btpb.ReadRowsRequest{
+			TableName: tbl,
+		},
+	}
+
+	resp, err := (*client).ReadRows(ctx, req)
+	if err != nil {
+		t.Fatalf("testproxy test: ReadRows returned error: %v", err)
+	}
+
+	if resp.Status.Code != int32(codes.OK) {
+		t.Errorf("testproxy test: ReadRows() didn't return OK; got %v", resp.Status.Code)
+	}
+
+	if len(resp.Row) != 1 {
+		t.Errorf("testproxy test: SampleRowKeys() returned wrong number of results; got: %d", len(resp.Row))
+	}
+}
+
+func TestMutateRow(t *testing.T) {
+	ctx := context.Background()
+	req := &pb.MutateRowRequest{
+		ClientId: tpc,
+		Request: &btpb.MutateRowRequest{
+			TableName: tbl,
+			RowKey:    []byte(rk),
+			Mutations: []*btpb.Mutation{
+				{
+					Mutation: &btpb.Mutation_SetCell_{
+						SetCell: &btpb.Mutation_SetCell{
+							ColumnQualifier: []byte("coll1"),
+							FamilyName:      "cf0",
+							Value:           []byte("mutant!"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	resp, err := (*client).MutateRow(ctx, req)
+	if err != nil {
+		t.Fatalf("testproxy test: MutateRow() returned error: %v", err)
+	}
+
+	if resp.Status.Code != int32(codes.OK) {
+		t.Errorf("testproxy test: MutateRow() didn't return OK; got %v", resp.Status.Code)
+	}
```

Comment:
Should this check that the new row has in fact mutated?

Reply:
Ditto below in BulkMutate

Reply:
Ensuring that the row was in fact mutated is delegated to the larger test suite operation. The code under test (the test proxy) doesn't directly mutate the row (that behavior happens in the Bigtable service, which is not under test).